### ISSUE TITLE
Feature Request - Filtering

### ISSUE THREAD

**Issue by danwalkercode** [jsimpso/PyFortiAPI#4]
Thanks for the work on PyFortiAPI, it's been a great base for me to get started with an internal project. Something I need to do, however, is perform queries for things like firewall policies beyond the inbuilt `name==<specific>`. 

I'll probably just fork and add on what I need, but I thought it'd be worth opening a feature request as it could be a useful feature for many. I'm not sure on what would be the cleanest way of doing it, I will likely add a filters parameter which is appended to the GET request.

**Comment by danwalkercode** [comment]
An example of what I'm doing:

```
# Firewall Policy Methods
    def get_firewall_policy(self, specific=False, filters=False):
        """
        Get firewall policy information from firewall

        :param specific: If provided, a specific object will be returned. If not, all objects will be returned.
            Specific can either be the policy name, or the policy ID.

        :return: JSON data for all objects in scope of request, nested in a list.
        """
        api_url = self.urlbase + "api/v2/cmdb/firewall/policy/"
        if specific:
            if type(specific) == int:
                api_url += str(specific)
            else:
                api_url += "?filter=name==" + specific
        elif filters:
            api_url += "?filter=" + filters
        results = self.get(api_url)
        if type(results) == int:
            return results
        elif len(results) == 0:
            return 404
        else:
            return results
```

Which means I can do:

```
policies = device.get_firewall_policy(filters="comments=@Search Pattern Here")
```

And just get the policies I need, searching against whatever field I want. This saves me having to load all policies and search through them with Python, which becomes a bit unfeasible when rulesets become enormous.

If this seems useful let me know and I'd be happy to do a PR for the changes + doc updates.

**Comment by jsimpso** [comment]
Hey @iDemonix ,

That sounds like a great addition - I'd be more than happy to merge that functionality in 👍

**Comment by jsimpso** [comment]
Merged PR #5
